src/server/bg_services/tier_ttf_worker.js

```diff
@@ -63,28 +64,34 @@ class TieringTTFWorker {
 
         const now = Date.now();
         for (const bucket of buckets) {
-            await node_allocator.refresh_tiering_alloc(bucket.tiering);
+            await node_allocator.refresh_tiering_alloc(bucket.tiering, this.last_run);
             const tiering_status = node_allocator.get_tiering_status(bucket.tiering);
             const selected_tier = mapper.select_tier_for_write(bucket.tiering, tiering_status);
-
-            const storage = await nodes_client.instance().aggregate_data_free_by_tier([String(selected_tier._id)],
-                selected_tier.system._id);
-            const tier_storage_free = size_utils.json_to_bigint(storage[String(selected_tier._id)][0].free);
-            const reports = await UsageReportStore.instance().get_usage_reports({
+            const storage = tiering_status[String(selected_tier._id)].mirrors_storage[0];
```

Comment:
As far as I know, the mirrors_storage has the following structure:
```
MIRROR = [
[SPREAD],
[SPREAD]...
]
``` 
When you are looking at index 0, you are disregarding the storage of the other spreads in the mirrors.
This means that you can have enough space in the first mirror but not enough spaces in all of the other mirrors and you won't do any cleaning job.
Which means that the whole cleaning job won't trigger when it should in case of mirrors that have more than one spread.

Reply:
Obviously, the whole algorithm beneath that checks the space and reports on it should change as well according to this.

Reply:
correct - fixed for the minimum space between the mirrors. as with map_server enough_room_in_tier.